### PR TITLE
GODRIVER-2514 Use MongoDB managed GCP account for GCP tests.

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary
- Use MongoDB managed GCP account for GCP tests.

# Background & Motivation
The integration tests added in DRIVERS-2377 were relying on a personal account.
[BUILD-15386](https://jira.mongodb.org/browse/BUILD-15386) provided a service account with necessary permissions.

Here is a patch build with the [testgcpkms tasks](https://spruce.mongodb.com/version/62f13d6f9ccd4e4612bed0eb/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2258,10 +2258,10 @@ task_groups:
           shell: "bash"
           script: |
             ${PREPARE_SHELL}
-            echo '${testgcpkms_key_file}' > /tmp/testgcpkms_key_file.json
+            echo '${testgcpkms_key_file_2}' > /tmp/testgcpkms_key_file.json
             export GCPKMS_KEYFILE=/tmp/testgcpkms_key_file.json
             export GCPKMS_DRIVERS_TOOLS=$DRIVERS_TOOLS
-            export GCPKMS_SERVICEACCOUNT="${testgcpkms_service_account}"
+            export GCPKMS_SERVICEACCOUNT="${testgcpkms_service_account_2}"
```
Comment (kevinAlbs) [inline .evergreen/config.yml:2264]: TODO: before merge, swap the project variables in https://spruce.mongodb.com/project/mongo-go-driver/settings/variables.

File: .evergreen/config.yml [hunk 2]
```diff
@@ -178,7 +178,7 @@ functions:
             # If this was a patch build, doing a fresh clone would not actually test the patch
             cp -R ${PROJECT_DIRECTORY}/ $DRIVERS_TOOLS
           else
-            git clone https://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
+            git clone https://github.com/kevinAlbs/drivers-evergreen-tools.git --branch D2377.update-creds $DRIVERS_TOOLS
```
Comment (kevinAlbs) [inline .evergreen/config.yml:181]: TODO: revert this change once https://github.com/mongodb-labs/drivers-evergreen-tools/pull/224 is merged.

**Review by prestonvasquez — APPROVED** [review]
LGTM!
